### PR TITLE
Signup: remove site information component and update site title

### PR DESCRIPTION
**Pull request by ramonjd**
### Using site title component in the flow

Since we're no longer capturing `phone` and `address` values, we don't need the site information component.

Instead, let's update the site title component with the latest styles, and plug its value into the state.

### Removing the site information component
We're also removing the site information component and all references to it (steps and flow refs as well)  in favour of the site title component.

### Retiring the site information Tracks event

We'll use `calypso_signup_actions_submit_site_title` in the place of `calypso_signup_actions_submit_site_information`.

👉 @megsfulton @niranjan-uma-shankar @fditrapani @adamziel You're using `calypso_signup_actions_submit_site_information` in some recent funnels I see. If this gets merged, that event will no longer fire. Let me know if there are any issues. Thanks!

## Testing instructions

Create a site and fill out the site title step. Check the following:

1. That we're saving `siteTitle` to the `state.signup.steps.siteTitle`
2. When you submit the site title step, we're firing `calypso_signup_actions_submit_site_title` tracking event with `{ value: 'your site title' }`
3. After creating a site, the site title you entered appears on your site.

Since #32432 we're not using the `site-information-without-domains` step, but just to make sure, test the flow using the `remove` variation of the `removeDomainsStepFromOnboarding` AB test –

Screen Shot 2019-04-25 at 2 36 39 pm

– and repeat the steps above. 👍 We've created site-title-without-domain to replace it.

### CODE REVIEW

**Comment by niranjan-uma-shankar** [comment]
> You're using calypso_signup_actions_submit_site_information in some recent funnels I see. 

Nothing that I'm using actively, feel free to go ahead.

**Comment by megsfulton** [comment]
>You're using calypso_signup_actions_submit_site_information in some recent funnels I see.

No issues for me. Thanks for checking.

**Comment by spen** [comment]
Hi @ramonjd, as I'm still a bit of a newbie to signup, only having poked around and made small changes in the past, I've jotted down some thoughts and questions while reviewing this PR. I figured I'd just get everything down that I wasn't sure about, so buckle up as this is a bit verbose!
I think there might be bits and pieces of signup that I'm unaware of, or some magic that's happening that I'm missing so I'm hoping you can clarify some of that.

I see that in `step-components.js` we're assigning `SiteTitleComponent` to both 
`'site-title-without-domains'` and `'site-title-with-preview'` steps, as well as the 'site-title' step

In `steps-pure.js`, we define those steps as so:
```js
'site-title': {
	stepName: 'site-title',
	providesDependencies: [ 'siteTitle' ],
},

'site-title-with-preview': {
	stepName: 'site-title-with-preview',
	providesDependencies: [ 'siteTitle' ],
	props: {
		showSiteMockups: true,
	},
},

'site-title-without-domains': {
	stepName: 'site-title-without-domains',
	apiRequestFunction: createSiteWithCart,
	delayApiRequestUntilComplete: true,
	dependencies: [ 'themeSlugWithRepo' ],
	providesDependencies: [ 'siteTitle', 'siteId', 'siteSlug', 'domainItem', 'themeItem' ],
	props: {
		showSiteMockups: true,
	},
},
```

From what I can gather, `with-preview` refers to the `showSiteMockups` prop being `true`.
So the difference between 'site-title' & 'site-title-with-preview' is quite obvious here - one shows the mockup the other doesn't. 
The differentiation 'site-title-without-domains' makes isn't so clear to me though:

I don't see `apiRequestFunction` being used in the component.
I don't see `delayApiRequestUntilComplete` being used in the component.
I don't see `themeSlugWithRepo` being used, though it's marked as a dependency - I'm assuming this still required though, in order for the `createSiteWithCart` call to work.
I see that the step states that it provides `siteTitle`, `siteId`, `siteSlug`, `domainItem` and `themeItem` as dependencies, but it looks like it's submitted with only `siteTitle`:
```js
// steps/site-title/index.#
SignupActions.submitSignupStep(
	{
		stepName,
		flowName,
	},
	{ siteTitle }
);
```
Are these extra provided dependencies a result of calling `createSiteWithCart`?

A lot of my questions I feel I've answered myself somewhat, but if there's anything above that I'm wrong on, please do let me know :)

**Comment by michaeldcain** [comment]
@spen: I'll try to answer some of your points - @ramonjd might want to correct me if/when I'm wrong. Largely, they're just related to the magic of Signup and code that hasn't been cleaned up. As you mention, you've pretty much answered the questions yourself :).

The `site-title-without-domains` step was added in #29494, it handles the validation and site creation that is typically handled by [the `domains` step](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/steps-pure.js#L214) while we run the domains removal AB test. As such, it's sort of a hybrid between the `site-title` and `domains` step.

The dependencies that are requested but not submitted explicitly through the step are generally provided by another step or via an API response. There's some more information about the [dependency/progress store here](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/README.md).

**Comment by ramonjd** [comment]
@spen Thanks for cogitation. I'm still doing a lot of head scratching myself. 

@cain summed it up well.

May I preface this comment with a 🥔 for the large PR. I tried to split it up, but the second PR would have been just deleting things and I wanted some sweet _iscalypsofast_ stats to print out and place on the toilet wall.

> I don't see themeSlugWithRepo being used, though it's marked as a dependency - I'm assuming this still required though, in order for the createSiteWithCart call to work.

I had to dig around to support my terrible memory, but selecting a site type (Business/Professional/etc) will [assign the value](https://github.com/Automattic/wp-calypso/blob/master/client/state/signup/steps/site-type/actions.js#L28), based on the values in [lib/signup/site-type.js ](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/site-type.js#L56)

File: client/signup/config/steps-pure.js [hunk 1]
```diff
@@ -444,60 +444,13 @@ export function generateSteps( {
...
-
-		'site-information-without-domains': {
-			stepName: 'site-information-without-domains',
+		'site-title-without-domains': {
+			stepName: 'site-title-without-domains',
 			apiRequestFunction: createSiteWithCart,
 			delayApiRequestUntilComplete: true,
 			dependencies: [ 'themeSlugWithRepo' ],
-			providesDependencies: [ 'title', 'siteId', 'siteSlug', 'domainItem', 'themeItem' ],
+			providesDependencies: [ 'siteTitle', 'siteId', 'siteSlug', 'domainItem', 'themeItem' ],
```
Comment (spen) [inline client/signup/config/steps-pure.js]: `title` ~> `siteTitle` is definitely an improvement! 👍

File: client/signup/steps/site-title/index.jsx [hunk 2]
```diff
@@ -3,85 +3,177 @@
...
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
+import Gridicon from 'gridicons';
```
Comment (spen) [inline client/signup/steps/site-title/index.jsx]: `gridicons` is external to Calypso, let's bump it up there!

**Pull request merged**